Prog/Cvd19_affctd_ind_pop.sas

```diff
@@ -0,0 +1,254 @@
+%DCData_lib( Ipums )
+
+%let RegPumas = "1100101", "1100102", "1100103", "1100104", "1100105", "2401600", "2400301", "2400302","2401001", "2401002", 
+				"2401003", "2401004", "2401005", "2401006", "2401007", "2401101", "2401102", "2401103", "2401104", "2401105", 
+				"2401106", "2401107", "5101301", "5101302", "5159301", "5159302", "5159303", "5159304", "5159305", "5159306", 
+				"5159307", "5159308", "5159309", "5110701", "5110702" , "5110703", "5151244", "5151245", "5151246", "5151255" ;
+
+%let keep_vars = 
+  serial pernum numprec perwt hhwt ind occ year statefip countyfip met2013 upuma
+  age sex 
+  raced racwht racblk racasian racamind racpacis racother hispand
+  educd school schltype
+  hud_inc poverty inc: 
+  labforce empstatd wkswork2 uhrswork tranwork classwkrd
+  ownershp ownershpd rentgrs owncost gq plumbing hotwater;
+
+data A;
+
+  set
+    Ipums.Acs_2014_18_dc (keep=&keep_vars)
+    Ipums.Acs_2014_18_md (keep=&keep_vars)
+    Ipums.Acs_2014_18_va (keep=&keep_vars)
+    Ipums.Acs_2014_18_wv (keep=&keep_vars);
+
+  where met2013 = 47900;
+  
+  retain total 1;
+
+  if upuma in (&RegPumas.) then MWCOG_region = 1;
+  else MWCOG_region = 0;
+  
+  /* Flag 35+ hours worked as full time */
+  if uhrswork >= 35 then fulltime=1;
+  else if uhrswork > 0 then fulltime=0;
+  else fulltime = .n;
+
+  /* Flag 50-52 weeks per year as year-round */
+  if wkswork2 = 6  then yearround=1;
+  else if wkswork2 > 0 then yearround=0;
+  else yearround = .n;
+
+  if 9920 > ind > 0 then do; 
+  
+    select;
+    
+      when ( year in ( 2018 ) )
+        if ind in ( 370:490, 6070:6090, 6180:6190, 6280, 7670, 8560:8590, 8660:8690 ) then cvd19_affctd_ind = 1;
+        else cvd19_affctd_ind = 0;
+    
+      when ( year in ( 2013:2017 ) )
+        if ind in ( 370:490, 6070:6090, 6180:6190, 6280, 7670, 8560:8590, 8660:8690 ) then cvd19_affctd_ind = 1;
+        else cvd19_affctd_ind = 0;
+        
+      when ( year in ( 2008:2012 ) )
+        if ind in ( 370:490, 6070:6090, 6180:6190, 6280, 7670, 8560:8590, 8660:8690 ) then cvd19_affctd_ind = 1;
+        else cvd19_affctd_ind = 0;
+        
+      otherwise do;
+        %err_put( msg="ACS industry not coded for this year. " year= )
+      end;
+      
+    end;
+    
+  end;
+  else cvd19_affctd_ind = .n;
+  
+  cvd19_affctd_incearn = cvd19_affctd_ind * incearn;
+  cvd19_affctd_incwage = cvd19_affctd_ind * incwage;
+  cvd19_affctd_incbus00 = cvd19_affctd_ind * incbus00;
+  
+  format MWCOG_region fulltime yearround cvd19_affctd_ind dyesno.;
+
+run;
+
+proc sort data=A;
+  by year serial pernum;
+run;
+
+proc freq data=A;
+  tables cvd19_affctd_ind;
+  tables cvd19_affctd_ind * ind / list nocum nopercent;
+run;
+
+proc means data=A n sum mean min max;
+  var inc: cvd19_affctd_inc: ;
+run;
+
+proc summary data=A;
+  by year serial;
+  var cvd19_affctd_ind inc: cvd19_affctd_inc: ;
+  output out=cvd19_affctd_ind_hh (drop=_type_ _freq_) sum= /autoname;
+run;
+
+
+data Covid19.cvd19_affctd_ind_pop;
```

Comment:
I had to switch this to Work.cvd19_affctd_ind_pop; for testing. Best practice is to use a %finalize_data_set and then move the proc tabulates at the bottom of this program to the other program.